bundles/core/org.eclipse.smarthome.core.id/src/main/java/org/eclipse/smarthome/core/id/UUID.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.id;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.config.core.ConfigConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class provides a unique ID for the instance that can be used for identification, e.g. when
+ * integrating with external systems. The UUID is generated only once and written to the file system, so that it does
+ * not change over time.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ */
+public class UUID {
```

Comment:
But it does more than the Java UUID class. It also persists the UUID. And it provides the specific UUID for the installation. Our Event class is more or less the same as the OSGi event. If I see a class just called UUID, I would not expect, that it holds the ID of the installation.